### ISSUE TITLE
Clarify where @id should not yet be provided / will be provided by server

### ISSUE THREAD

**Issue by cwebber** [w3c-social/activitypub#62]
In the case of activities wrapping existing objects, or when an object is either being posted for the first time with Create, an @id probably should not generally be provided by the client, because it's up to the server to determine that @id.  This isn't made clear in the current spec.  Probably, the server SHOULD ignore that @id and return its own.

It might be argued that in some server situations, the user maybe could provide the @id for the object being wrapped... I'm not sure.

**Comment by gobengo** [comment]
> Probably, the server SHOULD ignore that @id and return its own.

Better yet, return a 400 with an appropriate response body indicating the error

**Comment by rhiaro** [comment]
I was just about to file a similar issue to say that all of the example Activities currently don't have `id` (making them blank nodes) which would also make it very difficult for 

> Such deduplication MUST be performed by comparing the id of the activities and dropping any activities already seen. 

to happen.

Seems like it should just be explicit wrt who is required to do what with regards to generating `id`s for activities here, and made clear that ultimately all Activities MUST have `id`s.

I agree that when a _client_ is sending an Activity to a _server_ the server is ultimately responsible for generating the `id`. [LDP uses the HTTP `Slug` header](https://www.w3.org/TR/ldp/#ldpc-post-slug) to allow the client to hint, which is probably more useful for objects than activities, but you never know.

**Comment by cwebber** [comment]
Text is in the spec now about this, I think added by @rhiaro in fact:

```
        If an Activity is submitted with a value in the <code>id</code> 
        property, servers MUST ignore this and generate a new <code>id</code>
        for the Activity.
```

But yes about a MUST on `id`s.  I think we even agreed that transient activities/objects are possible using a fragment (or a redirect, if you want to provoke ire in certain semweb circles!).  I'll add that.

**Comment by cwebber** [comment]
Closed, see 95b596f for details.
